mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/ListeningCurrentQuotaUpdater.java

```diff
@@ -75,63 +75,66 @@ public boolean isHandling(Event event) {
-        computeQuotaOperation(expunged, quotaRoot).ifPresent(Throwing.<QuotaOperation>consumer(quotaOperation -> {
-            Mono.from(currentQuotaManager.decrease(quotaOperation))
-                .then(Mono.defer(Throwing.supplier(() -> eventBus.dispatch(
-                    EventFactory.quotaUpdated()
-                        .randomEventId()
-                        .user(expunged.getUsername())
-                        .quotaRoot(quotaRoot)
-                        .quotaCount(quotaManager.getMessageQuota(quotaRoot))
-                        .quotaSize(quotaManager.getStorageQuota(quotaRoot))
-                        .instant(Instant.now())
-                        .build(),
-                    NO_REGISTRATION_KEYS)).sneakyThrow()))
-                .block();
-        }).sneakyThrow());
+    private Mono<Void> handleExpungedEvent(Expunged expunged, QuotaRoot quotaRoot) {
+        return computeQuotaOperation(expunged, quotaRoot)
+            .flatMap(Throwing.<QuotaOperation, Mono<Void>>function(quotaOperation ->
+                Mono.from(currentQuotaManager.decrease(quotaOperation))
+                    .then(Mono.defer(Throwing.supplier(() -> eventBus.dispatch(
+                        EventFactory.quotaUpdated()
+                            .randomEventId()
+                            .user(expunged.getUsername())
+                            .quotaRoot(quotaRoot)
+                            .quotaCount(quotaManager.getMessageQuota(quotaRoot))
+                            .quotaSize(quotaManager.getStorageQuota(quotaRoot))
+                            .instant(Instant.now())
+                            .build(),
+                        NO_REGISTRATION_KEYS)).sneakyThrow()))).sneakyThrow());
     }
 
-    private void handleAddedEvent(Added added, QuotaRoot quotaRoot) {
-        computeQuotaOperation(added, quotaRoot).ifPresent(Throwing.<QuotaOperation>consumer(quotaOperation -> {
-            Mono.from(currentQuotaManager.increase(quotaOperation))
-                .then(Mono.defer(Throwing.supplier(() -> eventBus.dispatch(
-                    EventFactory.quotaUpdated()
-                        .randomEventId()
-                        .user(added.getUsername())
-                        .quotaRoot(quotaRoot)
-                        .quotaCount(quotaManager.getMessageQuota(quotaRoot))
-                        .quotaSize(quotaManager.getStorageQuota(quotaRoot))
-                        .instant(Instant.now())
-                        .build(),
-                    NO_REGISTRATION_KEYS)).sneakyThrow()))
-                .block();
-        }).sneakyThrow());
+    private Mono<Void> handleAddedEvent(Added added, QuotaRoot quotaRoot) {
+        return computeQuotaOperation(added, quotaRoot)
+            .flatMap(Throwing.<QuotaOperation, Mono<Void>>function(quotaOperation ->
+                Mono.from(currentQuotaManager.increase(quotaOperation))
+                    .then(Mono.defer(Throwing.supplier(() -> eventBus.dispatch(
```

Comment:
why supplier and not callable (that happens to allow Throwable)?

Reply:
I improved this code in follow up commits